pandas/core/arrays/timedeltas.py

```diff
@@ -394,6 +395,11 @@ def f(x):
 # Constructor Helpers
 
 def _generate_regular_range(start, end, periods, offset):
+    # TODO: Remove when _Day replaces Day
+    if hasattr(offset, '_prefix') and offset._prefix == 'D':
```

Comment:
i am not sure why you think warning on all creations like this is a good idea at all and it’s not what we discussed

this would be a nightmare for us and users

IIRC we need a very selective deprecation

Reply:
@jorisvandenbossche mentioned https://github.com/pandas-dev/pandas/issues/22864#issuecomment-425366667 that we need deprecation for these 4 points I mentioned https://github.com/pandas-dev/pandas/issues/22864#issue-364655882.

Essentially anywhere that 'D' is used as absolute day needs deprecating which hits a lot of tests.

Reply:
those are for arithmetic
you are deprecating creation which seems crazy

Reply:
We also have to depreciate `timedelta_range(..., freq='D')` since `'D'` will be relative and not absolute in favor of `'24H'`

Reply:
(This was a cumbersome point for me to tackle when I tried this the first time; that's why I opted for `'CD'` originally)